app/utils/logger.js

```diff
@@ -0,0 +1,58 @@
+// app/utils/logger.js
+
+const { createLogger, transports, format } = require('winston');
+const { combine, timestamp, printf, colorize, splat, simple } = format;
+const fs = require('fs');
+const path = require('path');
+require('winston-daily-rotate-file');
+
+// Ensure log directory exists
+const logDirectory = path.join(__dirname, '../../logs');
+fs.existsSync(logDirectory) || fs.mkdirSync(logDirectory);
```

Comment:
I am not sure about this line of code, normally the log package will handle all the directory creation stuff. Please double check the capability of the winston package.